controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -449,82 +439,129 @@ func (r *MultiClusterObservabilityReconciler) SetupWithManager(mgr ctrl.Manager)
 		Complete(r)
 }
 
+func checkStorageChanged(mcoOldConfig, mcoNewConfig *mcov1beta2.StorageConfig) {
+
+	if mcoOldConfig.AlertmanagerStorageSize != mcoNewConfig.AlertmanagerStorageSize {
+		isAlertmanagerStorageSizeChanged = true
+	}
+	if mcoOldConfig.CompactStorageSize != mcoNewConfig.CompactStorageSize {
+		isCompactStorageSizeChanged = true
+	}
+	if mcoOldConfig.RuleStorageSize != mcoNewConfig.RuleStorageSize {
+		isRuleStorageSizeChanged = true
+	}
+	if mcoOldConfig.ReceiveStorageSize != mcoNewConfig.ReceiveStorageSize {
+		isReceiveStorageSizeChanged = true
+	}
+	if mcoOldConfig.StoreStorageSize != mcoNewConfig.StoreStorageSize {
+		isStoreStorageSizeChanged = true
+	}
+}
+
 // HandleStorageSizeChange is used to deal with the storagesize change in CR
 // 1. Directly changed the StatefulSet pvc's size on the pvc itself for
 // 2. Removed StatefulSet and
 // wait for operator to re-create the StatefulSet with the correct size on the claim
 func (r *MultiClusterObservabilityReconciler) HandleStorageSizeChange(
 	mco *mcov1beta2.MultiClusterObservability) (*reconcile.Result, error) {
-	thanosPVCList := &corev1.PersistentVolumeClaimList{}
-	thanosPVCListOpts := []client.ListOption{
-		client.InNamespace(config.GetDefaultNamespace()),
-		client.MatchingLabels(map[string]string{
-			"app.kubernetes.io/instance": config.GetMonitoringCRName(),
-		}),
-	}
 
-	err := r.Client.List(context.TODO(), thanosPVCList, thanosPVCListOpts...)
-	if err != nil {
-		return &reconcile.Result{}, err
+	if isAlertmanagerStorageSizeChanged {
+		isAlertmanagerStorageSizeChanged = false
+		err := updateStorageSizeChange(r.Client,
+			map[string]string{
+				"observability.open-cluster-management.io/name": mco.GetName(),
+				"alertmanager": "observability",
+			}, mco.GetName(), mco.Spec.StorageConfig.AlertmanagerStorageSize)
```

Comment:
s/mco.Spec.StorageConfig.AlertmanagerStorageSize/mco.Spec.StorageConfig.ReceiveStorageSize/ ?

Reply:
so as to the items below.